mlir/utils/generateE2ETest.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+"""
+This script reads in a configuration file and generates MLIR E2E tests.
+
+The configuration files are in TOML format.  Below is an example:
+
+    directory= "mlir/test/mlir-miopen-driver/auto_e2e/"
+    prefix = "miopen-gen"
+    suffix = "%pv %random_data %xdlops | mlir-miopen-driver -c | mlir-rocm-runner --shared-libs=%rocm_wrapper_library_dir/librocm-runtime-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext --entry-point-result=void | FileCheck %s --check-prefix="
+
+    [[axis]]
+    name = "operation"
+    values = ["conv2d", "conv2d_bwd_weight", "conv2d_bwd_data"]
+    prefix = "--operation"
+
+    ## Resnet50
+    [[suite]]
+    name = "Resnet50"
+
+    [[suite.test]]
+    config = "-batchsize=256 -in_channels=1024 -in_h=14 -in_w=14 -out_channels=2048 -fil_h=1 -fil_w=1 --dilation_h=1 --dilation_w=1 --conv_stride_h=2 --conv_stride_w=2 --padding_h=0 --padding_w=0"
+
+    [[suite.test]]
+    config = "--more-options"
+    [[suite.test.exclude]]
+    name = "operation"
+    values = ["conv2d_bwd_data"]
+
+"""
+import tomli
+import itertools
+import os
+import sys
+import getopt
+
+def generate_option_list(prefixes: dict, table: list, key1: str, key2: str):
+    options_list=[]
+    for item in table[key1]:
+        options=[]
+        for value in item[key2]:
+            if item["name"] in prefixes:
+                opt = prefixes[item["name"]] + ' ' + value
+            else:
+                opt = value
+            options.append(opt)
+        options_list.append(options)
+
+    combinations=[]
```

Comment:
Random question - could we want this to be a `set()` 

Heck, you can probably do `combinations = frozenset(itertools.product(*options_list))`

Reply:
And IIRC set/dict have a stable iteration order these days